boards/arm/arduino_nano_33_ble/arduino_nano_33_ble_defconfig

```diff
@@ -19,3 +19,7 @@ CONFIG_BOOTLOADER_BOSSA_LEGACY=y
 
 # additional board options
 CONFIG_GPIO_AS_PINRESET=y
+# Following configuration needed to enable serial console via
+# USB-TTL UART
+CONFIG_BOARD_ARDUINO_NANO_33_BLE_EN_USB_CONSOLE=y
```

Comment:
It is just a side effect, that enables SERIAL + CONSOLE + UART_INTERRUPT_DRIVE and allows you to archive desired result. But 'select USB' and 'select USB_UART_CONSOLE' do not make sense if you connect "USB-TTL UART" to your board.

Reply:
@jfischer-no : Thanks a lot for review.
I see that without enabling CONFIG_USB_UART_CONSOLE=y, i cannot use USB-TTL for serial console. But when add CONFIG_USB_UART_CONSOLE=y, it has dependency on CONFIG_BOARD_ARDUINO_NANO_33_BLE_EN_USB_CONSOLE=y, error log as attached for your kind reference:
```
[0/1] Re-running CMake...
Including boilerplate (Zephyr base (cached)): /home/ohos-bld/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/ohos-bld/zephyrproject/zephyr/samples/philosophers
-- Zephyr version: 2.6.0-rc1 (/home/ohos-bld/zephyrproject/zephyr), build: v2.6.0-rc1-336-g457a28bf78a4
-- Found west (found suitable version "0.10.1", minimum required is "0.7.1")
-- Board: arduino_nano_33_ble
-- Cache files will be written to: /home/ohos-bld/.cache/zephyr
-- Using toolchain: zephyr 0.12.3 (/home/ohos-bld/zephyr-sdk-0.12.3)
-- Found dtc: /home/ohos-bld/zephyr-sdk-0.12.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/ohos-bld/zephyrproject/zephyr/boards/arm/arduino_nano_33_ble/arduino_nano_33_ble.dts
-- Generated zephyr.dts: /home/ohos-bld/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /home/ohos-bld/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /home/ohos-bld/zephyrproject/zephyr/build/zephyr/include/generated/device_extern.h
Parsing /home/ohos-bld/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/ohos-bld/zephyrproject/zephyr/boards/arm/arduino_nano_33_ble/arduino_nano_33_ble_defconfig'
Merged configuration '/home/ohos-bld/zephyrproject/zephyr/samples/philosophers/prj.conf'

warning: USB_CDC_ACM (defined at subsys/usb/class/Kconfig.cdc:4) has direct dependencies USB_DEVICE_STACK with value n, but is currently being y-selected by the following symbols:
 - USB_UART_CONSOLE (defined at drivers/console/Kconfig:105), with value y, direct dependencies CONSOLE (value: y), and select condition CONSOLE (value: y)

error: Aborting due to Kconfig warnings

CMake Error at /home/ohos-bld/zephyrproject/zephyr/cmake/kconfig.cmake:264 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/ohos-bld/zephyrproject/zephyr/cmake/app/boilerplate.cmake:555 (include)
  /home/ohos-bld/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /home/ohos-bld/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/ohos-bld/zephyrproject/zephyr/build/CMakeFiles/CMakeOutput.log".
See also "/home/ohos-bld/zephyrproject/zephyr/build/CMakeFiles/CMakeError.log".
FAILED: build.ninja 
/usr/bin/cmake --regenerate-during-build -S/home/ohos-bld/zephyrproject/zephyr/samples/philosophers -B/home/ohos-bld/zephyrproject/zephyr/build
ninja: error: rebuilding 'build.ninja': subcommand failed
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/ohos-bld/zephyrproject/zephyr/build
```

So, if we need to enable serial console via USB-TTL, then we need to enable these options. 

In case of Arduino, for many of the basic samples, USB/CDC_ACM is not present by default. It defiantly helps to have serial via USB-TTL UART.